code/implementation/docker/common/src/main/java/io/cattle/platform/docker/util/DockerUtils.java

```diff
@@ -33,4 +35,15 @@ public static Host getHostFromContainer(ObjectManager objectManager, Instance in
         return getHostFromContainer(objectManager, instance, DockerHostConstants.KIND_DOCKER);
     }
 
+    /**
+     * The docker API allows for identification of a container by name or id.
+     * This method will return the docker container id if it is available.
+     * If it isn't available, the rancher container uuid will be returned, since that is currently used
+     * as the name for containers that do not have external id set.
```

Comment:
I would change this so that it is more clear that external id given by docker to a container.